WordPress/WordPressTest/TestContextManager.m

```diff
@@ -30,90 +26,97 @@ - (instancetype)init
 
 - (NSManagedObjectModel *)managedObjectModel
 {
-    return _managedObjectModel ?: [super managedObjectModel];
+    return [_stack managedObjectModel];
 }
 
-- (NSPersistentStoreCoordinator *)persistentStoreCoordinator
+- (void)setManagedObjectModel:(NSManagedObjectModel *)managedObjectModel
 {
-    if (_persistentStoreCoordinator) {
-        return _persistentStoreCoordinator;
-    }
+    [_stack setManagedObjectModel:managedObjectModel];
+}
 
-    // This is important for automatic version migration. Leave it here!
-    NSDictionary *options = @{
-        NSInferMappingModelAutomaticallyOption          : @(YES),
-        NSMigratePersistentStoresAutomaticallyOption    : @(YES)
-    };
-
-    NSError *error = nil;
-    _persistentStoreCoordinator = [[NSPersistentStoreCoordinator alloc]
-                                   initWithManagedObjectModel:self.managedObjectModel];
-
-    if (![_persistentStoreCoordinator addPersistentStoreWithType:NSInMemoryStoreType
-                                                   configuration:nil
-                                                             URL:nil
-                                                         options:options
-                                                           error:&error]) {
-        NSLog(@"Unresolved error %@, %@", error, [error userInfo]);
-        abort();
-    }
+- (NSPersistentStoreCoordinator *)persistentStoreCoordinator
+{
+    return [_stack persistentStoreCoordinator];
+}
 
-    return _persistentStoreCoordinator;
+- (void)setPersistentStoreCoordinator:(NSPersistentStoreCoordinator *)persistentStoreCoordinator
+{
+    return [_stack setPersistentStoreCoordinator:persistentStoreCoordinator];
 }
 
 - (NSPersistentStoreCoordinator *)standardPSC
 {
-    return [super persistentStoreCoordinator];
+    return [_stack standardPSC];
 }
 
 - (NSManagedObjectContext *)mainContext
 {
-    if (_mainContext) {
-        return _mainContext;
-    }
+    return [_stack mainContext];
+}
 
-    _mainContext = [[NSManagedObjectContext alloc] initWithConcurrencyType:NSMainQueueConcurrencyType];
-    _mainContext.persistentStoreCoordinator = self.persistentStoreCoordinator;
+- (void)setMainContext:(NSManagedObjectContext *)mainContext
+{
+    [_stack setMainContext:mainContext];
+}
 
-    return _mainContext;
+-(void)setTestExpectation:(XCTestExpectation *)testExpectation
+{
+    [_stack setTestExpectation:testExpectation];
 }
 
 - (void)saveContext:(NSManagedObjectContext *)context
 {
     [self saveContext:context withCompletionBlock:^{
-        if (self.testExpectation) {
-            [self.testExpectation fulfill];
-            self.testExpectation = nil;
-        } else if (self.requiresTestExpectation) {
+        if (self->_stack.testExpectation) {
+            [self->_stack.testExpectation fulfill];
+            self->_stack.testExpectation = nil;
+        } else if (self->_stack.requiresTestExpectation) {
```

Comment:
Perhaps `self.` instead of the old school `->`?

Reply:
(Same note in the methods below!)

Reply:
I actually had to use `self->`, the compiler didn't liked the `self.` and I have no idea why

Reply:
Nevermind, done in 363c7fb1391760c0a5a27e38cfa1581860c03b12